test/unit/data-sources/rural-payments.test.js

```diff
@@ -8,59 +9,108 @@ const logger = {
 }
 
 describe('RuralPayments', () => {
-  test('handles RPP errors', () => {
-    // eslint-disable-next-line no-new
-    const rp = new RuralPayments({ logger })
-
-    const error = new Error('test error')
-    error.extensions = { response: { status: 400, headers: { get: () => 'text/html' } } }
-    const request = {}
-    const url = 'test url'
-
-    rp.didEncounterError(error, request, url)
-
-    expect(logger.error).toHaveBeenCalledWith('#datasource - Rural payments - request error', {
-      error,
-      request,
-      response: error.extensions.response,
-      code: RURALPAYMENTS_API_REQUEST_001
+  describe('fetch', () => {
+    const mockFetch = jest.spyOn(RESTDataSource.prototype, 'fetch')
+    const dummyRequest = { method: 'POST' }
+
+    beforeEach(() => {
+      mockFetch.mockClear().mockReset()
+    })
+    afterAll(() => {
+      mockFetch.mockRestore()
     })
-  })
 
-  test('handles APIM errors with error.extensions', () => {
-    // eslint-disable-next-line no-new
-    const rp = new RuralPayments({ logger })
+    test('returns data from RPP', async () => {
+      const rp = new RuralPayments({ logger })
 
-    const error = new Error('test error')
-    error.extensions = { response: { status: 400 } }
-    const request = {}
-    const url = 'test url'
+      mockFetch.mockResolvedValueOnce('data')
+
+      expect(await rp.fetch('path', dummyRequest)).toBe('data')
+      expect(mockFetch).toBeCalledTimes(1)
+    })
 
-    rp.didEncounterError(error, request, url)
+    describe('throws upstream errors from RPP', () => {
+      test('when the RPP service encounters an error', async () => {
+        const error = new Error('Server error')
+        error.extensions = { response: { status: 500 } }
 
-    expect(logger.error).toHaveBeenCalledWith('#datasource - apim - request error', {
-      error,
-      request,
-      response: error.extensions.response,
-      code: APIM_APIM_REQUEST_001
+        mockFetch.mockRejectedValue(error)
+
+        const rp = new RuralPayments({ logger })
+        try {
+          await rp.fetch('path', dummyRequest)
+        } catch (error) {
```

Comment:
Can we give the caught error a different name, as seems like it would conflict with the mock error you have defined.